govcd/disk_test.go

```diff
@@ -0,0 +1,391 @@
+
+	// Clean up
+	AddToCleanupList(fmt.Sprintf("%s|%s", disk.Disk.Name, disk.Disk.HREF), "disk", "", "Test_DiskRefresh")
+}
+
+// Test find disk attached VM
+func (vcd *TestVCD) Test_DiskAttachedVM(check *C) {
+
+	// Create disk
+	diskCreateParamsDisk := &types.Disk{
+		Name:        vcd.config.VCD.Disk.Name,
+		Size:        vcd.config.VCD.Disk.Size,
+		Description: vcd.config.VCD.Disk.Description,
+	}
+
+	diskCreateParams := &types.DiskCreateParams{
+		Disk: diskCreateParamsDisk,
+	}
+
+	disk, err := vcd.vdc.CreateDisk(diskCreateParams)
+	check.Assert(err, IsNil)
+	check.Assert(disk.Disk.Name, Equals, diskCreateParamsDisk.Name)
+	check.Assert(disk.Disk.Size, Equals, diskCreateParamsDisk.Size)
+	check.Assert(disk.Disk.Description, Equals, diskCreateParamsDisk.Description)
+
+	// Wait for disk creation complete
+	task := NewTask(vcd.vdc.client)
+	for _, taskItem := range disk.Disk.Tasks.Task {
+		task.Task = taskItem
+		err = task.WaitTaskCompletion()
+		check.Assert(err, IsNil)
+	}
+
+	// Find VM
+	vmType, _ := vcd.find_first_vm(vcd.find_first_vapp())
+	vm := NewVM(&vcd.client.Client)
+	vm.VM = &vmType
+
+	// Attach disk
+	attachDiskTask, err := vm.AttachDisk(&types.DiskAttachOrDetachParams{
+		Disk: &types.Reference{
+			HREF: disk.Disk.HREF,
+		},
+	})
+	check.Assert(err, IsNil)
+
+	err = attachDiskTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	// Get attached VM
+	vmRef, err := disk.AttachedVM()
+	check.Assert(err, IsNil)
+	check.Assert(vmRef, NotNil)
+	check.Assert(vmRef.Name, Equals, vm.VM.Name)
+
+	// Clean up
+	AddToCleanupList(fmt.Sprintf("%s|%s", disk.Disk.Name, disk.Disk.HREF), "disk", "", "Test_DiskAttachedVM")
+}
+
+// Test find Disk by Href in VDC struct
+func (vcd *TestVCD) Test_VdcFindDiskByHREF(check *C) {
+
+	// Create disk
+	diskCreateParamsDisk := &types.Disk{
+		Name:        vcd.config.VCD.Disk.Name,
+		Size:        vcd.config.VCD.Disk.Size,
+		Description: vcd.config.VCD.Disk.Description,
+	}
+
+	diskCreateParams := &types.DiskCreateParams{
+		Disk: diskCreateParamsDisk,
+	}
+
+	disk, err := vcd.vdc.CreateDisk(diskCreateParams)
+	check.Assert(err, IsNil)
+	check.Assert(disk.Disk.Name, Equals, diskCreateParamsDisk.Name)
+	check.Assert(disk.Disk.Size, Equals, diskCreateParamsDisk.Size)
+	check.Assert(disk.Disk.Description, Equals, diskCreateParamsDisk.Description)
+
+	// Wait for disk creation complete
+	task := NewTask(vcd.vdc.client)
+	for _, taskItem := range disk.Disk.Tasks.Task {
+		task.Task = taskItem
+		err = task.WaitTaskCompletion()
+		check.Assert(err, IsNil)
+	}
+
+	// Find disk by href
+	foundDisk, err := vcd.vdc.FindDiskByHREF(disk.Disk.HREF)
+	check.Assert(err, IsNil)
+	check.Assert(disk.Disk, NotNil)
+	check.Assert(disk.Disk.Name, Equals, foundDisk.Disk.Name)
+
+	// Clean up
+	AddToCleanupList(fmt.Sprintf("%s|%s", disk.Disk.Name, disk.Disk.HREF), "disk", "", "Test_VdcFindDiskByHREF")
+}
+
+// Test find disk by href and vdc client
+func (vcd *TestVCD) Test_FindDiskByHREF(check *C) {
+	// Create disk
```

Comment:
If the parameters needed for disk are not found (and in our case, after my comments above, it should be only the disk size) the test should be skipped.
Therefore, every disk related test should check whether the necessary parameter(s) are provided, and skip if they aren't.